### DISCUSSION TITLE
v2.0

### DISCUSSION THREAD

**Discussion by rdbende** [rdbende/Sun-Valley-ttk-theme#47] [Announcements]
## What's changed
- Merge the control images into a single file for better performance. Many thanks @TransparentLC for the idea and implementation ([#30](https://github.com/rdbende/Sun-Valley-ttk-theme/issues/30))
- Add keyboard navigation support 🎉
- Fix UI resizing issue with Treeview widget. Huge thanks to the Tk dev team for coming up with a brilliant solution to this bug ([ticket](https://core.tcl-lang.org/tk/tktview?name=bc602049ab))
- Fix Scale when setting sticky
- Small positioning fixes on Combobox, Spinbox and Scrollbar widgets
- Better panedwindow sash
- Fix incorrect font name by @sumeshir26 in ([#15](https://github.com/rdbende/Sun-Valley-ttk-theme/pull/15))

## New Contributors
* @sumeshir26 made their first contribution in https://github.com/rdbende/Sun-Valley-ttk-theme/pull/15
* @TransparentLC made their first contribution in https://github.com/rdbende/Sun-Valley-ttk-theme/issues/30

<hr /><em>This discussion was created from the release <a href='https://github.com/rdbende/Sun-Valley-ttk-theme/releases/tag/v2.0'>v2.0</a>.</em>

**Comment by sumeshir26** [comment]
Does the new sprite sheet affect performance?
And can this easily be ported to GIF? If yes, can I work on it?

**Reply by rdbende** [reply]
> Does the new sprite sheet affect performance?

Yes, in a good way. Startup time is faster because it now uses only 2 png files, whereas before it had to open and read 188 images one by one.

> And can this easily be ported to GIF?

Probably yes, though I abandoned that idea because it was complicated to make and it looked ugly.
If you want to work on it, you are very welcome!

**Reply by TransparentLC** [reply]
I guess the port can be simply done because only the output format needs to be changed. Note that GIF image has a 256-color palette limitation, so you have to make a sprite sheet for the light theme and another one for the dark theme. You shouldn't combine these two themes into only one image, otherwise unwanted dithering will be applied which reduces the image's quality.

**Comment by XiangQinxi** [comment]
我觉得有必要解决一下打包的问题了
I think it is necessary to solve the problem of packaging
![image](https://user-images.githubusercontent.com/90026267/194236331-ea433674-8e7b-4499-9125-c5819c3f7dea.png)

**Reply by TransparentLC** [reply]
如果是使用 PyInstaller 的话，在打包的命令里需要加上 `--collect-data sv_ttk`，spec 文件需要加上：

```python
from PyInstaller.utils.hooks import collect_data_files

block_cipher = None

a = Analysis(
    datas=[collect_data_files('sv_ttk')], # <-- 加上这个
    ...
)
pyz = PYZ(a.pure, a.zipped_data, cipher=block_cipher)

...
```

Add `--collect-data sv_ttk` if you are using PyInstaller.

**Reply by rdbende** [reply]
This was explained in #38. I have now pinned it, so others can more easily find info about it.
